src/math.cpp

```diff
@@ -246,8 +250,14 @@ auto ceil(const CallContext& ctx) -> Value {
         .location = ctx.call_location(),
         .reverse = [](const Value& new_value,
                       const Value& old_value) -> std::optional<SourceChangeTree> {
-            return old_value.force(new_value); // Just guessing that this is the right value because
-                                               // every information about post-comma numbers is lost
+            Number n = std::get<Number>(new_value);
+            if (std::fmod(n.value, 1.0) == 0.0) {
+                return old_value.force(
+                    new_value); // Just guessing that this is the right value because
+                                // every information about post-comma numbers is lost
```

Comment:
```suggestion
                // Just guessing that this is the right value because every
                // information about post-comma numbers is lost
                return old_value.force(new_value); 
```

clang-format sometimes does some strange formatting when there are comments at the end of the line.